editor/index.js

```diff
@@ -12,6 +19,10 @@ import './assets/stylesheets/main.scss';
 import Layout from './layout';
 import { createReduxStore } from './state';
 
+// Configure moment globally
+moment.locale( settings.l10n.locale );
```

Comment:
The question is: if we do this in `data/index.js`, `wp-data` won't be library independent anymore. If you look at the code, you'll notice it's already done there, temporary but reverted to the original locale.